### PR TITLE
Upgrade ply to 3.8

### PR DESCRIPTION
**Pull request by doctaphred**
Got bitten by a bug today that's fixed in ply 3.6, so I figured I'd share the fix, which is just upgrading the bundled version of ply.

Travis builds succeeded, but I confess I couldn't get one test to pass on my machine (OS X, Python 2.7), either before or after the changes. Stack trace is below in case you're curious, but it's a separate issue. Let me know if you want more information, but otherwise I won't worry about it.

Let me know if I can do anything else to help. I'm new to open source, so apologies for any rookie mistakes. Cheers, and thanks for the great work!

```
ERROR: test_with_cpp (test_general.TestParsing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fwagner/dev/pycparser/tests/test_general.py", line 31, in test_with_cpp
    cpp_args='-I%s' % c_files_path)
  File "./pycparser/__init__.py", line 94, in parse_file
    return parser.parse(text, filename)
  File "./pycparser/c_parser.py", line 147, in parse
    debug=debuglevel)
  File "./pycparser/ply/yacc.py", line 265, in parse
    return self.parseopt_notrack(input,lexer,debug,tracking,tokenfunc)
  File "./pycparser/ply/yacc.py", line 1047, in parseopt_notrack
    tok = self.errorfunc(errtoken)
  File "./pycparser/c_parser.py", line 1700, in p_error
    column=self.clex.find_tok_column(p)))
  File "./pycparser/plyparser.py", line 55, in _parse_error
    raise ParseError("%s: %s" % (coord, msg))
ParseError: /Users/fwagner/dev/pycparser/tests/c_files/memmgr.c:1:1: before: /
```

### CODE REVIEW

File: pycparser/ply/__init__.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # PLY package
 # Author: David Beazley (dave@dabeaz.com)
 
+__version__ = '3.7'
```
Comment (doctaphred) [inline pycparser/ply/__init__.py]: Not sure why this still says 3.7, but it's legit: https://github.com/dabeaz/ply/blob/master/ply/__init__.py#L4

**Comment by eliben** [comment]
Thanks, couple of questions/requests:

* Please update the README.rst of pycparser which mentions PLY's version
* Can you specify somewhere (maybe in the README) where you downloaded PLY exactly (link) - because of the version confusion
* Finally once all's done, please squash the PR to a single commit so I can merge

**Comment by doctaphred** [comment]
Done! How does this look?

**Comment by eliben** [comment]
Thanks!

**Pull request merged**